### ISSUE TITLE
[BUG] Rancher Chart UI doesn't respect the value in charts/questions.yaml

### ISSUE THREAD

**Issue by PhanLe1010** [rancher/dashboard#8079]
**Rancher Server Setup**
- Rancher version: v2.7.1
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): Docker
- Proxy/Cert Details: SelfCert 

**Information about the Cluster**
- Kubernetes version: v1.24.9 (rke2)
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Digital Ocean

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom): Admin/Cluster 

**Describe the bug**
Rancher Chart UI doesn't respect the value in `charts/questions.yaml`, it always uses the value in `charts/values.yaml`

**To Reproduce**
1. Deploy Rancher v2.7.1
2. Create Kubernetes v1.24.9+rke2r2 cluster
3. Add the chart repo `https://github.com/rancher/charts.git` with the branch `dev-v2.7`
4. Navigate to install app from that repo
5. Obverse that the installation UI doesn't respect the value in `charts/questions.yaml`, it always uses the value in `charts/values.yaml`. For example, this Longhorn chart has `enablePSP` set to `true` in [questions.yaml](https://github.com/rancher/charts/blob/d9748b1cc8a5299ea6a153e4d695c69e615593da/charts/longhorn/101.2.0%2Bup1.4.0/questions.yaml#L827) and `false` in [values.yaml](https://github.com/rancher/charts/blob/d9748b1cc8a5299ea6a153e4d695c69e615593da/charts/longhorn/101.2.0%2Bup1.4.0/values.yaml#L321). The Rancher UI picks the `false` value ( see the attached picture)

![Screenshot from 2023-02-01 14-25-59](https://user-images.githubusercontent.com/22139961/216178859-a2cd0804-9f23-489d-8bc5-514388221806.png)

**Expected Result**
The installation UI respect the value in `charts/questions.yaml`

**Additional context**
This is probably why Longhorn QA (when testing with Kubernetes 1.25) doesn't catch the mistake that the `enablePSP` is still set to `true` inside `questions.yaml` even though Kubernetes 1.25 doesn't support PSP. It was because Rancher UI doesn't use that value.

**Comment by jordojordo** [comment]
This is intended behavior as issue stems from the value of [`enablePSP`](https://github.com/rancher/charts/blob/d9748b1cc8a5299ea6a153e4d695c69e615593da/charts/longhorn/101.2.0%2Bup1.4.0/values.yaml#L321)  being defined in `values.yaml`. [This condition](https://github.com/rancher/dashboard/blob/b78b3cc61b8aeb6b3cfec9ec7bc542284a190782/shell/components/Questions/Question.js#L69-L71) will check to see if the variable is `undefined` and if the variable in the `questions.yaml` is defined, if so it would then [apply the value](https://github.com/rancher/dashboard/blob/b78b3cc61b8aeb6b3cfec9ec7bc542284a190782/shell/components/Questions/index.vue#L481) from the questions.

If you're able to change the value in `values.yaml` to be undefined or have it match the chart value this should cause the checkbox to be enabled.

**Comment by PhanLe1010** [comment]
Hi @jordojordo 

Does it mean that the data in values.yaml gets higher priority than the value in questions.yaml? Should the design be the other way around? The reason is that usually the values.yaml come from the upstream chart, then when building the rancher chart we create questions.yaml to define a layer on top of values.yaml. Should the UI prioritize the values in questions.yaml?

**Comment by jordojordo** [comment]
@PhanLe1010 According to [our documentation](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/helm-charts-in-rancher/create-apps#question-variable-reference) the `default` property should only be used if there is no corresponding value in the `values.yml`. I'm unsure of how the other charts with questions will behave if we prioritize the value from those questions, this may lead to unintended issues.

**Comment by gaktive** [comment]
@PhanLe1010 this sounds like it's working as expected since this is an established pattern that other components use. Is there a reason why two different values appear in the YAML files?

Passing to QA to validate though it's more Longhorn to then tell UI if there's something else to consider.

**Comment by gaktive** [comment]
We're touching some parts of the logic for release components in https://github.com/rancher/dashboard/issues/8034

**Comment by PhanLe1010** [comment]
Thanks @jordojordo and @gaktive for the explanation. 

If this is expected behavior, we can close the issue. 

The 2 values were different because the values from values.yaml comes from upstream chart. We set the values in question.yaml when we build Rancher chart. Sometime, the values in upstream chart got updated but the dev forgot to update the value in questions.yaml. However, QA couldn't catch the mistake because the behavior overhere is that the values.yaml overwrite the questions.yaml file. I personal feel it should be the other way around (i.e., the questions.yaml overwrite the values.yaml)
